From 53f05f5d275d211deebcea7be2c3b762bcc041bd Mon Sep 17 00:00:00 2001
From: Kevin Grandemange <kevin.grandemange@allegrodvt.com>
Date: Mon, 15 Oct 2018 10:22:02 +0200
Subject: [PATCH 1/4] spacing (uncrustify instability)

Signed-off-by: Kevin Grandemange <kevin.grandemange@allegrodvt.com>
---
 common/al_codec.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/common/al_codec.c b/common/al_codec.c
index 997f9fb..97a7107 100644
--- a/common/al_codec.c
+++ b/common/al_codec.c
@@ -320,7 +320,6 @@ int al5_codec_release(struct inode *inode, struct file *filp)
 			dev_err(codec->device,
 				"Failed to destroy channel on mcu. Something went wrong");
 
-
 		/* best effort. If everything went wrong, still free the channel resources
 		 * to avoid leaks */
 		al5_user_destroy_channel_resources(user);
-- 
2.7.4

